src/wg/format/png/reader.d

```diff
@@ -0,0 +1,926 @@
+module wg.format.png.reader;
+
+import wg.util.allocator;
+import wg.format.png.types;
+import wg.util.util: Result, asArrayOf;
+import wg.image.imagebuffer;
+
+// TODO: 
+// 1. Make CommonMetadata using pHYs chunk data
+// 2. Make PngChunks smarter so they keep their allocator and chunks can easily be added and removed
+// 3. Check sRGB, gAMA and cHRM chunks and return the proper format string using them
+
+private extern(C) nothrow @nogc
+{
+    import etc.c.zlib : z_stream, z_streamp, ZLIB_VERSION;
+
+    // Had to copy function definitions from zlib here in order to add @nogc to them
```

Comment:
should probably make a PR to phobos

Reply:
Done: https://github.com/dlang/phobos/pull/6973